iohk/Github.hs

```diff
@@ -0,0 +1,52 @@
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-implicit-prelude -Wno-semigroup #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Github where
+
+import           Data.Aeson           (FromJSON, parseJSON, withObject, (.:))
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.Text            as T
+import           GHC.Generics         (Generic)
+import           GHC.Stack            (HasCallStack)
+import           System.Directory     (doesFileExist)
+import           Utils                (fetchJson')
+
+type Rev = T.Text
+type Org = T.Text
+type Repo = T.Text
+
+data CommitStatus = CommitStatus {
+    state      :: T.Text
+    , statuses :: [ Status ]
+    , sha      :: T.Text
+    } deriving (Show, Generic)
+instance FromJSON CommitStatus
```

Comment:
You guys knows this probably better than me, but using `Generic` deriving for JSON is implicitly binding ourselves to a specific generic-derived format for our JSON. If we want to be a bit more agnostic, then indeed using `deriveFromJSON` from `Data.Aeson.TH` might be better.